### PR TITLE
[Distributed] enable tensor_parallel_output for finetuning

### PR DESCRIPTION
**Pull request by SylarTiaNII**
### PR types

Bug fixes(Performance optimization)
### PR changes

Others
### Description

enable tensor_parallel_output as default for better performance

### CODE REVIEW

File: llm/finetune_generation.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def main():
             # NOTE(gongenlei): new add autotuner_benchmark
             model_config = AutoConfig.from_pretrained(
                 model_args.model_name_or_path,
-                tensor_parallel_output=False,
+                tensor_parallel_output=True,
```
Comment (wawltor) [inline llm/finetune_generation.py]: tensor_parallel_output=True主要是为了加速吗？ 

tensor_parallel_output=True  设置为True会导致模型指标ACC计算出错，因为没有对结果进行all gather操作

Comment (SylarTiaNII) [reply llm/finetune_generation.py]: 如果不设置为True的话，一个是影响性能，再一个是影响显存占用。在llm场景会有比较大的性能影响。模型指标ACC的计算是不是可以考虑做一下相应的优化来适配mp场景？

Comment (ZHUI) [reply llm/finetune_generation.py]: 这里可以加开关，建议默认还是False，generation 那里还没有适配，会有问题。

File: paddlenlp/trainer/trainer.py [hunk 2]
```diff
@@ -2780,6 +2780,12 @@ def evaluation_loop(
 
         # Metrics!
         if self.compute_metrics is not None and all_preds is not None and all_labels is not None:
+            if self.args.tensor_parallel_degree > 1 and all_preds.shape != all_labels.shape:
```
Comment (ZHUI) [inline paddlenlp/trainer/trainer.py]:
```suggestion
            if self.args.tensor_parallel_degree > 1 and isinstance(all_preds, paddle.Tensor) all_preds.shape != all_labels.shape:
```

然后这里加一个注释吧，all_gather logits for tp

Comment (lugimzzz) [inline paddlenlp/trainer/trainer.py]: 应该在https://github.com/PaddlePaddle/PaddleNLP/blob/develop/llm/utils.py#L208 CausalLMTrainer这里对logit加上all gather的操作，而不是在这

**Review by wawltor — APPROVED** [review]
LGTM
